docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,642 @@
+			</div>
+		);
+	}
+}
+```
+
+### Passing extra classes to widgets
+
+The theming mechanism provides a simple way to consistently apply custom styles across every widget in an application, but isn't flexible enough for scenarios where a user wants to apply additional styles to specific instances of a given widget.
+
+Extra styling classes can be passed in through a [themeable widget's `classes` property](#themedmixin-widget-properties). They are considered additive, and do not override the widget's existing styling classes - their purpose is instead to allow fine-grained tweaking of pre-existing styles. Each set of extra classes provided need to be grouped by two levels of keys:
+
+1.  The appropriate [widget theme key](#widget-theme-keys), specifying the widget that the classes should be applied to, including those for any child widgets that may be utilized.
+2.  Specific existing CSS classes that the widget utilizes, allowing widget consumers to target styling extensions at the level of individual DOM elements, out of several that a widget may output.
+
+For illustration, the type definition for the extra classes property is:
+
+```ts
+type ExtraClassName = string | null | undefined | boolean;
+
+interface Classes {
+	[widgetThemeKey: string]: {
+		[baseClassName: string]: ExtraClassName[];
+	};
+}
+```
+
+As an example of providing extra classes, the following tweaks an instance of a [Dojo combobox](https://github.com/dojo/widgets/tree/master/src/combobox), as well as the [text input](https://github.com/dojo/widgets/tree/master/src/text-input) child widget it contains. This will change the background color to blue for both the text input control used by the combobox as well as its control panel. The down arrow within the combo box's control panel will also be colored red:
+
+> src/styles/MyComboBoxStyleTweaks.m.css
+
+```css
+.blueBackground {
+	background-color: blue;
+}
+
+.redArrow {
+	color: red;
+}
+```
+
+> src/widgets/MyWidget.tsx
+
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+import ComboBox from '@dojo/widgets/combobox';
+
+import * as myComboBoxStyleTweaks from '../styles/MyComboBoxStyleTweaks.m.css';
+
+const myExtraClasses = {
+	'@dojo/widgets/combobox': {
+		controls: [myComboBoxStyleTweaks.blueBackground],
+		trigger: [myComboBoxStyleTweaks.redArrow]
+	},
+	'@dojo/widgets/text-input': {
+		input: [myComboBoxStyleTweaks.blueBackground]
+	}
+};
+
+export default class MyWidget extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				Hello from a tweaked Dojo combobox!
+				<ComboBox classes={myExtraClasses} results={['foo', 'bar']} />
+			</div>
+		);
+	}
+}
+```
+
+Note that it is a widget author's responsibility to explicitly pass the `classes` property to all child widgets that are leveraged, as the property will not be injected nor automatically passed to children by Dojo itself.
+
+## Making themeable applications
```

Comment:
Do we think this should be higher up in the order? This feels like it's the first thing that people will want to do.

Reply:
I originally had it near the top, but decided to be consistent with the ordering we established for i18n, i.e. first set up all the pieces, then plug them together.

That said, theming may be different in that while there are no 'default' message bundles, apps could use an out-the-box theme without needing to author anything themselves.

Would expected usage here override a desire for consistency across documentation?